pelican/plugins/photos/photos.py

```diff
@@ -1907,8 +1892,35 @@ def detect_inline_galleries(content: Union[Article, Page]):
             pelican_settings["PHOTO_INLINE_GALLERY_PATTERN"], content._content
         )
         for m in gallery_strings:
+            gallery_name = m.group("gallery_name")
+            options_strg = m.group("options")
+            options = {}
+            if options_strg is not None:
+                options_list = re.compile(r"[\^,]+").split(options_strg)
+                r_pair = re.compile(r"(?P<variable>\w+)=(?P<value>\w+)")
+                r_flag = re.compile(r"(?P<flag>\w+)")
+                for opt in options_list:
+                    m_pair = r_pair.match(opt)
+                    m_flag = r_flag.match(opt)
+                    if m_pair:
+                        d = m_pair.groupdict()
+                        value = d["value"]
+                        # convert string to bool when possible
+                        if value == "True":
+                            value = True
+                        elif value == "False":
+                            value = False
+                        options[d["variable"]] = value
+                    elif m_flag:
+                        d = m_flag.groupdict()
+                        options[d["flag"]] = True
+                    else:
+                        logger.error(
+                            f"unexpected inline gallery <options> = {options_strg}"
+                        )
```

Comment:
Can you split this parsing out into a function? It would be easier to test (and if you put a test in for it, that'd be grand, too)